### ISSUE TITLE
Sometimes monsters fail to draw cards?

### ISSUE THREAD

**Issue by benknoble** [benknoble/frosthaven-manager#43]
This happened once when launching the app, but restarting it fixed the problem.

During a game, an existing empty group stopped drawing cards after new monsters were added. Similarly, a group that was added mid-scenario stopped drawing cards (but only after going from empty to non-empty, I think).

**Comment by benknoble** [comment]
1. Load the state below (save it to a file carefully, since there are… odd… bytes).
2. Run `frosthaven-manager <file>`.
3. Kill the existing archer with the button.
4. Click "Next Round."
5. Add any numbered archer via the button.
6. Click "Draw Abilities."

You'll even notice that, if you save this broken state and deserialize it, the `ability-decks-current` of `(hash-ref (@! (state-@ability-decks)) "Archer")` is not `#f`. Loading this new saved state produces a "working" view.

---

Here's the serialized state:
```
racket/fasl:Äåq|libfrosthaven-manager/defns.rktdeserialize-info:money-v0deserialize-info:material-kinddeserialize-info:herb-kinddeserialize-info:material-v0deserialize-info:herb-v0!deserialize-info:random-item-type!deserialize-info:monster-modifier$frosthaven-manager/manager/state.rktstate-deserialize-infodeserialize-info:creature-v0deserialize-info:player-v0"deserialize-info:monster-group*-v0!deserialize-info:monster-group-v0deserialize-info:monster-v0,frosthaven-manager/manager/ability-decks.rkt!deserialize-info:ability-decks-v0„unonpnqooprqoooonoooppoppoppopoopoopoopnopq?tsqqur|q}wqtxq}xsnr|qqwq}yqtuq}{rrqq{qqsqtvtvtqtttptnScoutAlgox ScoutImpairPriestAlgox Priest	Shield  1	Pierce  2	Shield  2	Pierce  3GuardAlgox Guardf
monster-statsuqqzqr
City GuardRetaliate  2	Shield  3Retaliate  3HoundRetaliate  1ArcherAlgox ArcherspqvprMove -1Move +0Nothing SpecialHeal 3, Range 3Calculated StrikeMove +1	Attack +1Shield 1
Hasty Assault	Attack -1	Attack +0:Attack +0 (+2 if target adjacent to any of Hound's allies)Nothing Special 1Nothing Special 2Move -2Attack -1, Pierce 2Attack +1, Range 3monster-abilityìRancid Arrow§c∏Attack +0, Range 3, Poisonì≤ñØ≥ìGreedë
Move +1, JumpLoot 1ì¨£´¥ì
Noxious Blade Attack +2, Poisonì	Cruel Bowã™Attack -1, Range 3, ImpairìRapid BoltsΩAttack -1, Target 2, Range 4ìÆ≥™∞ñTend to the Clan 1Ö´≠ñ
Divine Energy«™Heal 1, all adjacent allies
Bless selfñ¨¨´Attack +0, Range 3ñDrain Strengthv´Attack -1, Range 3, DisarmñTend to the Clan 2Ö´≠ñDrain Speedv™Attack +0, Range 3, Immobilizeñ	Angry HexwØ#Attack -1, Target 2, Range 3, CurseñÆ∏™∫ú∂†´¥úParry and Thrust}±Retaliate 2úPsych Up•™¥Strengthen selfú
Venom Shiv}Attack +0, Poison±ú≤åØ≥ú∑†´¥úÆ¥™∞úThrowing Axeë™Attack +0, Range 2§Harrowing Howlu´Muddle all adjacent§∂à´¥§Pack Hunting 2Å´µ§Gripping Teetht™Attack +0, Immobilize§Pack Hunting 1Å´µ§Æ¡∏∞§Sharp Fangs∂π∏π§∑à´¥¶
Shoot Footã´Attack -1, Range 5, Immobilize¶Close Iné´∫¶¨ç´Attack +0, Range 4¶Æö™Attack +1, Range 4¶
Power ShotÆAttack +1, Range 5¶
Twin Bolts¶≥Target 2, Range 4¶Set Trap|™Attack -1, Range 5FCreate one 3-damage trap in an adjacent empty hex closest to an enemy.¶≤~ØAttack -1, Range 4uplayorvpw	u
Boneshaperttnnvow	
Blinkbladevvnnvqw	Geminatevvnnvnw	Friggxxnnvrxoy¶ßo®©zosvsxyúùoûü

h-equal(nnnnnnnnnnnnooooooppråqqvÜqqxqqyqqzãrärroÑo~àçqtwÇqtyqtzqt{rÄÖpq}sq}uq}vÅÉáq}zâr(~nÄÅÇÉÑÖonÜáàâpäãåçÇ
unfused




}(étréètsêètuéêéèêtoêééêèèëëëëëëëëëëíííííííííí
ì
îmonster-infoìîuqpvrpyrpzrq}rqÄsqÜsrèsrxrqzrq{srïsrïÑsrïàssïéttïñtuïñ
óñóspptppòupqòwpqòzpqò}pqòÇpròâpròuqqôuqqöôwqqöôzqqöô|qröôÅqröôáqsöõêqsöõú
ùúùtpqûxqqzqr}rrÅrrársèssxprü}qsÅrsÑrtâstìsuùtu†ú†sppsppòuppòvpqòwqqòxqqö{qröqrötpqòtpqöwpqöwprö•xqrö°{qrö°}qr¢£Çqs¢£§
§§§rqprrp•trp•vrp•wrq•xrq°|sq°sr°tsptsp°usq°vsr°ysr°{sr£ts£Ött£¶
ß¶ßroq®upquprxpr{prÅprÑpstpr©yprzps~psÄptäptèpu
ì¿ºæª¬ø¡Ωñ»∆≈√« ƒ…úÃ—“ŒœÀ–Õ§÷”◊’‘⁄ÿŸ¶€·‹‚ﬂ›ﬁ‡
ì{ªºΩæø¿¡¬ñ{√ƒ≈∆«»… ú{ÀÃÕŒœ–—“§{”‘’÷◊ÿŸ⁄¶{€‹›ﬁﬂ‡·‚
```

**Comment by benknoble** [comment]
@bogdanp, if you have time, I wouldn't mind if you could confirm the analysis in 198bd6a. It does seem like the obvious problem based on my memory of our conversations about `cond-view`, and I think it's solved the problem.

**Comment by Bogdanp** [comment]
I haven't looked very deeply, but the commit message sounds plausible and the changes look good.

Makes me wonder if we can extend `racket/gui/easy/debugger` to highlight these sorts of issues...

**Comment by benknoble** [comment]
Thanks for looking!

**Comment by Bogdanp** [comment]
I was trying to take another look at this today coming from https://github.com/Bogdanp/racket-gui-easy/issues/73#issuecomment-3819892894. Unfortunately, the save no longer works today due to an arity error. Trying an old commit doesn't work either, because qi has made breaking changes since. Any chance you could share a save that gets me in the state to reproduce the original issue when you have some time @benknoble ?

**Comment by benknoble** [comment]
> I was trying to take another look at this today coming from [Bogdanp/racket-gui-easy#73 (comment)](https://github.com/Bogdanp/racket-gui-easy/issues/73#issuecomment-3819892894). Unfortunately, the save no longer works today due to an arity error. Trying an old commit doesn't work either, because qi has made breaking changes since. Any chance you could share a save that gets me in the state to reproduce the original issue when you have some time [@benknoble](https://github.com/benknoble) ?

@Bogdanp I appreciate you looking at this again; I was able to make the save file run on Frosthaven Manager d4db313 (gui: draw new ability cards mid-round if necessary, 2023-03-15) (the commit prior to the fix), using drym-org/qi@d2d8e50e (bump version 2.0 -> 3.0, 2022-07-29) since I have Qi installed as a link to the source: `racket -y gui/manager.rkt save` did it for me.

However, and this is a bit ironic, I could not reproduce the original issue. I didn't try downgrading gui-easy to see if that was related, though; `raco pkg show` says I currently have bogdanp/racket-gui-easy@8f2ac018ff6c2c553bf9529a08e2acfea1c995fa. Even going back to Frosthaven Manager 813ede2 (dev: make it easier to launch a save, 2023-03-07), which is a few commits after the one I blamed in the fix (to include the ability to launch saves from the command line), I still can't reproduce the original issue. Odd.

**Comment by benknoble** [comment]
PS I _expect_ the original to no longer be reproducible today, anyway.

Trying to follow my original train of thoughts based on this text

> You'll even notice that, if you save this broken state and deserialize it, the ability-decks-current of (hash-ref (@! (state-@ability-decks)) "Archer") is not #f. Loading this new saved state produces a "working" view.

It seems that I was able to find out that some observable state in the broken view `#f`, but the actual state was correct (whence my ideas about the views being destroyed and not seeing updates?).
